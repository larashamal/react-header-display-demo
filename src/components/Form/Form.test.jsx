/**
 * TODO:
 * 1. Write a snapshot test for Form component.
 * 2. Write a test for the Form component's handleSubmit function. Use a mock fxn. and confirm that the setName function is called whenever the form is submitted.
 * HINT: `userEvent.click` to trigger a form submission.
 * HINT: You will only submit the form once, so only check that the mock function is called once.
 */

import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import Form from "./Form";

it("renderers without crashing", () => {
  const rendered = renderer.create(<Form setName={() => {}} />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it("calls the submit handler when the form is submitted", () => {
  const handleSubmit = jest.fn();

  render(<Form setName={handleSubmit} />);
  const submitBtn = screen.getByRole("button", { name: /submit/i });

  userEvent.click(submitBtn);
  expect(handleSubmit).toHaveBeenCalled();
});
