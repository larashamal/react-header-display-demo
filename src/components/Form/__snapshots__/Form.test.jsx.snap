// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderers without crashing 1`] = `
<form
  className="mx-auto flex w-48 flex-col items-center"
  onSubmit={[Function]}
>
  <div>
    <label
      className="sr-only"
      htmlFor="name"
    >
      Name
    </label>
    <input
      className="rounded border pl-2"
      id="name"
      name="formName"
      placeholder="Enter your name"
      type="text"
    />
  </div>
  <button
    className="mt-4 rounded bg-green-500 px-4 py-2 text-white"
    type="submit"
  >
    Submit
  </button>
</form>
`;
